.vagrant/Vagrantfile

```diff
@@ -0,0 +1,196 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# NOTES:
+# This version was tested using Windows 11 Home Edition
+# It uses the "vagrant-gatling-rsync" Vagrant plugin to sync host folder to guest folder
+# It uses the "vagrant-rsync-back" plugin to sync guest folder to host folder
+# The sync from host to guest happens automatically on "vagrant up" after a "vagrant halt"
+# Once the sync is complete, rsync will monitor the host folder for changes and sync them to the guest folder.
+# To sync from guest to host, you must manually run "vagrant rsync-back" in a terminal window in vagrant project folder
+# Please install these plugins before issuing "vagrant up" by running the 2 commands in a terminal window.
+# vagrant plugin install vagrant-gatling-rsync
+# vagrant plugin install vagrant-rsync-back
+
+# Customize the following constants to suit your needs
+HOST_FOLDER = "../gloss-translation"
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+  # The most common configuration options are documented and commented below.
+  # For a complete reference, please see the online documentation at
+  # https://docs.vagrantup.com.
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://vagrantcloud.com/search.
+  config.vm.box = "generic/ubuntu2204"
+
+  # Configure for application API
+  config.vm.network "forwarded_port", guest: 4300, host: 4300, host_ip: "127.0.0.1"
+
+  # Configure for application Web
+  config.vm.network "forwarded_port", guest: 4200, host: 4200, host_ip: "127.0.0.1"
```

Comment:
I added the following line:

```ruby
config.vm.network "forwarded_port", guest: 5432, host: 5432, host_ip: "127.0.0.1"
```
however, when I tested using PgAdmin, I got the following error;

```text
connection failed: :1), 
port 5432 failed: could not receive data from server: Socket is not connected (0x00002749/10057) could not send SSL negotiation packet: Socket is not connected (0x00002749/10057)
```

I was able to resolve this problem by modifying the following postgres files: `postgresql.conf` and `pg_hba.conf`.  
To implement this changes, I would need to add this logic to the Vagrantfile dbsetup provisioning section.

2 Questions:
1. Do you want me to implement this and remove SSH Tunnelling instructions?
2. Do you want me to keep port at default 5432, or change to something else (5435?) just in case user has postgres running on host?

Reply:
Let's modify the config files and remove the ssh tunneling

Reply:
Done.

Reply:
Could we include an item in the "Caveat" section that mentions that Postgres running on the host needs to be stopped before starting the VM? I had postgres running on my host, and got this error
```
Vagrant cannot forward the specified ports on this VM, since they
would collide with some other application that is already listening
on these ports. The forwarded port to 5432 is already in use
on the host machine.
```
I stopped the postgres Windows service, and then the virtual machine ran fine. Maybe we should include a note about this in the docs, in case a future dev runs into the same error message.

Reply:
Add this as point 3. in the Caveat section:

1. If you have postgres database running on your host machine at default port `5432`, ensure to shut it down before bring up your VM machine. This can be done by stopping the postgres Windows service. This is because postgres in your VM also listens at the same default port number. Failure to do so will result in the following error:

   ```text
   Vagrant cannot forward the specified ports on this VM, since they
   would collide with some other application that is already listening
   on these ports. The forwarded port to 5432 is already in use
   on the host machine.
   ```

Reply:
Thank you!